spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,385 @@
+import static org.apache.iceberg.expressions.Expressions.notNull;
+import static org.apache.iceberg.expressions.Expressions.or;
+import static org.apache.iceberg.expressions.Expressions.startsWith;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.NaNUtil;
+import org.apache.spark.sql.connector.expressions.LiteralValue;
+import org.apache.spark.sql.connector.expressions.NamedReference;
+import org.apache.spark.sql.connector.expressions.filter.And;
+import org.apache.spark.sql.connector.expressions.filter.Not;
+import org.apache.spark.sql.connector.expressions.filter.Or;
+import org.apache.spark.sql.connector.expressions.filter.Predicate;
+import org.apache.spark.unsafe.types.UTF8String;
+
+public class SparkV2Filters {
+
+  private static final Pattern BACKTICKS_PATTERN = Pattern.compile("([`])(.|$)");
+
+  private SparkV2Filters() {}
+
+  private static final String TRUE = "ALWAYS_TRUE";
+  private static final String FALSE = "ALWAYS_FALSE";
+  private static final String EQ = "=";
+  private static final String EQ_NULL_SAFE = "<=>";
+  private static final String GT = ">";
+  private static final String GT_EQ = ">=";
+  private static final String LT = "<";
+  private static final String LT_EQ = "<=";
+  private static final String IN = "IN";
+  private static final String IS_NULL = "IS_NULL";
+  private static final String NOT_NULL = "IS_NOT_NULL";
+  private static final String AND = "AND";
+  private static final String OR = "OR";
+  private static final String NOT = "NOT";
+  private static final String STARTS_WITH = "STARTS_WITH";
+
+  private static final Map<String, Expression.Operation> FILTERS =
+      ImmutableMap.<String, Expression.Operation>builder()
+          .put(TRUE, Expression.Operation.TRUE)
+          .put(FALSE, Expression.Operation.FALSE)
+          .put(EQ, Expression.Operation.EQ)
+          .put(EQ_NULL_SAFE, Expression.Operation.EQ)
+          .put(GT, Expression.Operation.GT)
+          .put(GT_EQ, Expression.Operation.GT_EQ)
+          .put(LT, Expression.Operation.LT)
+          .put(LT_EQ, Expression.Operation.LT_EQ)
+          .put(IN, Expression.Operation.IN)
+          .put(IS_NULL, Expression.Operation.IS_NULL)
+          .put(NOT_NULL, Expression.Operation.NOT_NULL)
+          .put(AND, Expression.Operation.AND)
+          .put(OR, Expression.Operation.OR)
+          .put(NOT, Expression.Operation.NOT)
+          .put(STARTS_WITH, Expression.Operation.STARTS_WITH)
+          .build();
+
+  private static final int FIRST_ORDINAL = 0;
+  private static final int SECOND_ORDINAL = 1;
```

Comment:
I know I suggested first_ordinal and second_ordinal, but if you’re not in love with those, after looking at it further I think “left child” and “right child” might make sense too.

But I also don’t see any need to bike shed / change further just for the sake of changing. This is plenty readable as-is for me. Just throwing it out there as another idea.

Reply:
I took @aokolnychyi's suggestion to use `leftChild` and `rightChild`